--- a/src/board/last5.c
+++ b/src/board/last5.c
@@ -38,9 +38,3 @@
-    register __vec2x32float__ xy;
-    register float z;
-
-    asm {
-        psq_l xy, 0(src), 0, 0
-        lfs z, 8(src)
-        psq_st xy, 0(dst), 0, 0
-        stfs z, 8(dst)
-    }
+    dst->x = src->x;
+    dst->y = src->y;
+    dst->z = src->z;
@@ -66 +60 @@
-void mbObjHookReset(int modelId);
+void mbObjHookReset(s16 modelId);
