--- a/src/REL/fileseldll/filename.c
+++ b/src/REL/fileseldll/filename.c
@@ -28 +28 @@
-extern inline float sqrtf(float x)
+extern inline float mp6_fileseldll_sqrtf(float x)
@@ -580 +580 @@
-    len = sqrtf(dirX * dirX + dirY * dirY);
+    len = mp6_fileseldll_sqrtf(dirX * dirX + dirY * dirY);
@@ -589 +589 @@
-            dist = sqrtf(px * px + py * py);
+            dist = mp6_fileseldll_sqrtf(px * px + py * py);
@@ -592 +592 @@
-            angDist = sqrtf((px - ndx) * (px - ndx) + (py - ndy) * (py - ndy));
+            angDist = mp6_fileseldll_sqrtf((px - ndx) * (px - ndx) + (py - ndy) * (py - ndy));
