--- a/src/game/mic.c
+++ b/src/game/mic.c
@@ -13,0 +14,2 @@
+#include "game/dvd.h"
+#include "game/frand.h"
@@ -45 +47 @@
-static void MCExtHandler(void);
+static void MCExtHandler(s32 chan, s32 result);
@@ -64,0 +67 @@
+int HuMCResponseGet2(void);
@@ -306 +309 @@
-    return;
+    return 0;
@@ -955,3 +958,5 @@
-void MCExtHandler(void)
-
-{
+static void MCExtHandler(s32 chan, s32 result)
+
+{
+  (void)chan;
+  (void)result;
@@ -2363 +2368 @@
-  __src = (void *)HuDvdDataRead(param_1);
+  __src = HuDvdDataRead((char *)param_1);
